pkg/api/user.go

```diff
@@ -68,6 +68,11 @@ func (hs *HTTPServer) getUserUserProfile(c *contextmodel.ReqContext, userID int6
 		userProfile.AuthLabels = append(userProfile.AuthLabels, authLabel)
 		userProfile.IsExternal = true
 		userProfile.IsExternallySynced = login.IsExternallySynced(hs.Cfg, authInfo.AuthModule)
+		oAuthExternallySynced := false
+		if oauthInfo := hs.SocialService.GetOAuthInfoProvider(strings.TrimPrefix(authInfo.AuthModule, "oauth_")); oauthInfo != nil {
+			oAuthExternallySynced = oauthInfo.AllowAssignGrafanaAdmin
+		}
```

Comment:
note: annoyingly, I couldn't pass in the whole `oauthInfo` or `SocialService`, as that creates a circular dependency.